transport/src/main/java/io/netty/bootstrap/ChannelInitializerExtensions.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2023 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.bootstrap;
+
+import io.netty.util.internal.SystemPropertyUtil;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.ServiceLoader;
+
+/**
+ * The configurable facade that decides what {@link ChannelInitializerExtension}s to load and where to find them.
+ */
+abstract class ChannelInitializerExtensions {
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(ChannelInitializerExtensions.class);
+    private static volatile ChannelInitializerExtensions implementation;
+
+    private ChannelInitializerExtensions() {
+    }
+
+    /**
+     * Get the configuration extensions, which is a no-op implementation by default,
+     * or a service-loading implementation if the {@code io.netty.bootstrap.extensions} system property is
+     * {@code serviceload}.
+     */
+    static ChannelInitializerExtensions getExtensions() {
+        ChannelInitializerExtensions impl = implementation;
+        if (impl == null) {
+            synchronized (ChannelInitializerExtensions.class) {
+                impl = implementation;
+                if (impl != null) {
+                    return impl;
+                }
+                String extensionProp = SystemPropertyUtil.get("io.netty.bootstrap.extensions");
+                logger.debug("-Dio.netty.bootstrap.extensions: {}", extensionProp);
+                if ("serviceload".equalsIgnoreCase(extensionProp)) {
```

Comment:
If you prefer to keep system property:

PR description says that the expected value is boolean and I like that bcz users don't need to guess what value to pass and we don't need to document. However, the property should be renamed to something that will make it clear that the boolean value is expected. Consider: `io.netty.bootstrap.enableExtensions` or smth similar.